### PR TITLE
fix(): fix sidenav scroll subscription

### PR DESCRIPTION
**Pull request by kara**

### CODE REVIEW

File: src/app/shared/table-of-contents/table-of-contents.ts [hunk 1]
```diff
@@ -67,13 +67,17 @@ export class TableOfContents implements OnInit {
...
-
-    this._scrollSubscription = Observable
-      .fromEvent(this._scrollContainer, 'scroll')
-      .debounceTime(10)
-      .subscribe(() => this.onScroll());
+    // On init, the sidenav content element doesn't yet exist, so it's not possible
+    // to subscribe to its scroll event until next tick (when it does exist).
+    Promise.resolve().then(() => {
+      this._scrollContainer = this.container ?
+        this._document.querySelectorAll(this.container)[0] : window;
```
Comment (crisbeto) [inline src/app/shared/table-of-contents/table-of-contents.ts]: You can use `querySelector` instead of `querySelectorAll(...)[0]`.

Comment (kara) [reply src/app/shared/table-of-contents/table-of-contents.ts]: I'm just copy-pasting what was there before into a Promise, but I can update if you want.

File: src/app/shared/table-of-contents/table-of-contents.ts [hunk 2]
```diff
@@ -67,13 +67,17 @@ export class TableOfContents implements OnInit {
...
-    this._scrollContainer = this.container ?
-      this._document.querySelectorAll(this.container)[0] : window;
-
-    this._scrollSubscription = Observable
-      .fromEvent(this._scrollContainer, 'scroll')
-      .debounceTime(10)
-      .subscribe(() => this.onScroll());
+    // On init, the sidenav content element doesn't yet exist, so it's not possible
+    // to subscribe to its scroll event until next tick (when it does exist).
+    Promise.resolve().then(() => {
```
Comment (crisbeto) [inline src/app/shared/table-of-contents/table-of-contents.ts]: Couldn't you do this in one of the other lifecycle hooks?

Comment (kara) [reply src/app/shared/table-of-contents/table-of-contents.ts]: Fair point. I can move to `ngAfterViewInit`.

Comment (kara) [reply src/app/shared/table-of-contents/table-of-contents.ts]: Edit: this appears not to work.

**Pull request merged**